src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/AdjournHearingsWindowMappingTest.java

```diff
@@ -0,0 +1,185 @@
+            .build();
+        LocalDate result = AdjournHearingsWindowMapping.getHearingWindowStart(caseData);
+
+        assertThat(result).isEqualTo(expected);
+    }
+
+    @Nested
+    class ErrorCase {
+
+        @DisplayName("Throws exception when invalid hearingDateType")
+        @Test
+        void throwsExceptionWhenInvalidHearingDateType() {
+            SscsCaseData caseData = SscsCaseData.builder()
+                .adjournCaseNextHearingDateType("invalidType")
+                .build();
+
+            assertExceptionWhenInvalidData(caseData,
+                "Unexpected adjournCaseNextHearingDateType: invalidType");
+        }
+
+        @DisplayName("Throws exception when invalid dateOrPeriod")
+        @Test
+        void throwsExceptionWhenInvalidDateOrPeriod() {
+            SscsCaseData caseData = SscsCaseData.builder()
+                .adjournCaseNextHearingDateType(FIRST_AVAILABLE_DATE_AFTER)
+                .adjournCaseNextHearingDateOrPeriod("invalidValue")
+                .build();
+
+            assertExceptionWhenInvalidData(caseData,
+                "Unexpected adjournCaseNextHearingDateOrPeriod: invalidValue");
+        }
+
+        @DisplayName("Throws exception when missing dateOrPeriod")
+        @Test
+        void throwsExceptionWhenMissingDateOrPeriod() {
+            assertExceptionWithFirstAvailableDateAfterWhenMissingProperty(
+                null,
+                "adjournCaseNextHearingDateOrPeriod"
+            );
+        }
+
+        @DisplayName("Throws exception when missing dateAfterPeriod")
+        @Test
+        void throwsExceptionWhenMissingDateAfterPeriod() {
+            assertExceptionWithFirstAvailableDateAfterWhenMissingProperty(
+                PROVIDE_PERIOD,
+                "adjournCaseNextHearingFirstAvailableDateAfterPeriod"
+            );
+        }
+
+        @DisplayName("Throws exception when missing dateAfterDate")
+        @Test
+        void throwsExceptionWhenMissingDateAfterDate() {
+            assertExceptionWithFirstAvailableDateAfterWhenMissingProperty(
+                PROVIDE_DATE,
+                "adjournCaseNextHearingFirstAvailableDateAfterDate"
+            );
+        }
+
+        @DisplayName("Throws exception when invalid date")
+        @Test
+        void throwsExceptionWhenInvalidDate() {
+            SscsCaseData caseData = SscsCaseData.builder()
+                .adjournCaseNextHearingDateType(FIRST_AVAILABLE_DATE_AFTER)
+                .adjournCaseNextHearingDateOrPeriod(PROVIDE_DATE)
+                .adjournCaseNextHearingFirstAvailableDateAfterDate("not a date")
+                .build();
+
+            assertThatThrownBy(() -> AdjournHearingsWindowMapping.getHearingWindowStart(caseData))
+                .isInstanceOf(DateTimeParseException.class);
+        }
+
+        private void assertExceptionWithFirstAvailableDateAfterWhenMissingProperty(
+            String dateOrPeriod,
+            String missingProperty
+        ) {
```

Comment:
This can be put on the end of the line before (theres a few like these so I'd check the whole file)

Reply:
I like to put  ) { on a new line so there is clear separation with the indent

Reply:
My disappointment is immeasurable and my day is ruined.